core/events.js

```diff
@@ -347,879 +353,6 @@ Blockly.Events.fromJson = function(json, workspace) {
-  var location = {};
-  var parent = block.getParent();
-  if (parent) {
-    location.parentId = parent.id;
-    var input = parent.getInputWithBlock(block);
-    if (input) {
-      location.inputName = input.name;
-    }
-  } else {
-    location.coordinate = block.getRelativeToSurfaceXY();
-  }
-  return location;
-};
-
-/**
- * Does this event record any change of state?
- * @return {boolean} True if something changed.
- */
-Blockly.Events.Move.prototype.isNull = function() {
-  return this.oldParentId == this.newParentId &&
-      this.oldInputName == this.newInputName &&
-      goog.math.Coordinate.equals(this.oldCoordinate, this.newCoordinate);
-};
-
-/**
- * Run a move event.
- * @param {boolean} forward True if run forward, false if run backward (undo).
- */
-Blockly.Events.Move.prototype.run = function(forward) {
-  var workspace = this.getEventWorkspace_();
-  var block = workspace.getBlockById(this.blockId);
-  if (!block) {
-    console.warn("Can't move non-existant block: " + this.blockId);
-    return;
-  }
-  var parentId = forward ? this.newParentId : this.oldParentId;
-  var inputName = forward ? this.newInputName : this.oldInputName;
-  var coordinate = forward ? this.newCoordinate : this.oldCoordinate;
-  var parentBlock = null;
-  if (parentId) {
-    parentBlock = workspace.getBlockById(parentId);
-    if (!parentBlock) {
-      console.warn("Can't connect to non-existant block: " + parentId);
-      return;
-    }
-  }
-  if (block.getParent()) {
-    block.unplug();
-  }
-  if (coordinate) {
-    var xy = block.getRelativeToSurfaceXY();
-    block.moveBy(coordinate.x - xy.x, coordinate.y - xy.y);
-  } else {
-    var blockConnection = block.outputConnection || block.previousConnection;
-    var parentConnection;
-    if (inputName) {
-      var input = parentBlock.getInput(inputName);
-      if (input) {
-        parentConnection = input.connection;
-      }
-    } else if (blockConnection.type == Blockly.PREVIOUS_STATEMENT) {
-      parentConnection = parentBlock.nextConnection;
-    }
-    if (parentConnection) {
-      blockConnection.connect(parentConnection);
-    } else {
-      console.warn("Can't connect to non-existant input: " + inputName);
-    }
-  }
-};
-
-/**
- * Class for a block drag event. Fired when block dragged into or out of
- * the blocks UI.
- * @param {Blockly.Block} block The moved block.  Null for a blank event.
- * @extends {Blockly.Events.Abstract}
- * @constructor
- */
-Blockly.Events.DragOutside = function(block) {
-  if (!block) {
-    return;  // Blank event to be populated by fromJson.
-  }
-  Blockly.Events.DragOutside.superClass_.constructor.call(this, block);
-};
-goog.inherits(Blockly.Events.DragOutside, Blockly.Events.Abstract);
-
-/**
- * Class for a block drag event.
- * @param {Blockly.Block} block The moved block.  Null for a blank event.
- * @extends {Blockly.Events.Abstract}
- * @constructor
- */
-Blockly.Events.BlockDragOutside = Blockly.Events.DragOutside;
```

Comment:
This extra layer got removed in the refactor, I assume that's okay?

Reply:
Yes.  The extra layer exists for the block events because they used to be called by their short forms (e.g. Create instead of BlockCreate).  When we added more events (VarCreate, etc.) we renamed the block events to be clearer.  But we couldn't get rid of the old classes in case people were relying on them.

These new events don't have people who would be relying on them having short names, so no need for the indirection.